guides/03_building-with-blocks/06_theming-guide

```diff
@@ -0,0 +1,177 @@
+
+#### Referencing Core Variables
+
+To reference one of the core constructor variables, precede the variable name with an asterisk. To reference a core color, use the `*primary_`, `*secondary_`, or `*neutral_` prefix, followed by the brightness value. For example:
+
+```python
+theme = gr.themes.Default(primary_color="blue").set(
+    button_primary_background="*primary_200",
+    button_primary_background_hover="*primary_300",
+)
+```
+
+In the example above, we've set the `button_primary_background` and `button_primary_background_hover` variables to `*primary_200` and `*primary_300`. These variables will be set to the 200 and 300 brightness values of the blue primary color palette, respectively.
+
+Similarly, to reference a core size, use the `*spacing_`, `*radius_`, or `*text_` prefix, followed by the size value. For example:
+
+```python
+theme = gr.themes.Default(radius_size="md").set(
+    button_primary_border_radius="*radius_xl",
+)
+```
+
+In the example above, we've set the `button_primary_border_radius` variable to `*radius_xl`. This variable will be set to the `xl` setting of the medium radius size range.
+
+#### Referencing Other Variables
+
+Variables can also reference each other. For example, look at the example below:
+
+```python
+theme = gr.themes.Default().set(
+    button_primary_background="#FF0000",
+    button_primary_background_hover="#FF0000",
+    button_primary_border="#FF0000",
+)
+```
+
+Having to set these values to a common color is a bit tedious. Instead, we can reference the `button_primary_background` variable in the `button_primary_background_hover` and `button_primary_border` variables, using a `*` prefix. 
+
+```python
+theme = gr.themes.Default().set(
+    button_primary_background="#FF0000",
+    button_primary_background_hover="*button_primary_background",
+    button_primary_border="*button_primary_background",
+)
+```
+
+Now, if we change the `button_primary_background` variable, the `button_primary_background_hover` and `button_primary_border` variables will automatically update as well.
+
+This is particularly useful if you intend to share your theme - it makes it easy to modify the theme without having to change every variable.
+
+Note that dark mode variables automatically reference each other. For example:
+
+```python
+theme = gr.themes.Default().set(
+    button_primary_background="#FF0000",
+    button_primary_background_dark="#AAAAAA",
+    button_primary_border="*button_primary_background",
+    button_primary_border_dark="*button_primary_background",
+)
+```
+
+`button_primary_border_dark` will draw its value from `button_primary_background_dark`, because dark mode always draw from the dark version of the variable.
```

Comment:
There are a lot of theme properties that are never mentioned here. Some of them, like `body_background` are really exciting and open the room for a lot of creativity. I wonder if we could add a final section like: "A Full Custom Theme" that sets a bunch of different properties to give people an idea of what is really possible with themes.